Std/Data/String/Basic.lean

```diff
@@ -42,3 +42,13 @@ If `pre` is a prefix of `s`, i.e. `s = pre ++ t`, return the remainder `t`.
 -/
 def dropPrefix? (s : String) (pre : String) : Option Substring :=
   s.toSubstring.dropPrefix? pre.toSubstring
+
+/-- `s.stripPrefix p` will remove `p` from the beginning of `s` if it occurs there,
+or otherwise return `s`. -/
+def stripPrefix (s p : String) :=
```

Comment:
```suggestion
def stripPrefix (s p : String) : String :=
```

Reply:
Also I think it would be better to return an option string here and leave the decision making for what to do when the prefix isn't present to the user via `getD` et al.

Reply:
mathlib4 also defines:

```
/-- `isPrefixOf? pre s` returns `some post` if `s = pre ++ post`.
If `pre` is not a prefix of `s`, it returns `none`. -/
def isPrefixOf? (pre s : String) : Option String :=
  if startsWith s pre then some <| s.drop pre.length else none

/-- `isSuffixOf? suff s` returns `some pre` if `s = pre ++ suff`.
If `suff` is not a suffix of `s`, it returns `none`. -/
def isSuffixOf? (suff s : String) : Option String :=
  if s.endsWith suff then some <| s.dropRight suff.length else none
```
for cases where you want to do the decision making yourself.

I find though that I want the "just get rid of it" functions frequently "inline" with more string processing functions, rather like you often see a `.trim` in the middle of a string pipeline.

Perhaps moving these `Option` values functions to `Std4` at the same time would be a good solution?

Reply:
Oh, `Std4` already has `dropPrefix? : String \to String \to Option Substring`, so I've pushed a commit using that.

We don't have `dropSuffix?` yet. I added an inefficient implementation, and if you'd be happy with this PR otherwise, I can replace it with the efficient one.

Reply:
It seems like there should be a symmetric implementation for `dropSuffix?` since, assuming the string actually contains the substring, you know that `s.utf8Length - suff.utf8Length` is a valid `String.Pos` and can split there.

Reply:
Instead I factored out `Substring.eqAsString`, and implemented everything in terms of that. I *think* it is as performant, but you should be skeptical of that claim. :-) Happy to have a better name, too.

Reply:
Isn't `Substring.eqAsString` basically a special case of [`String.substrEq`](https://github.com/leanprover/lean4/blob/master/src/Init/Data/String/Basic.lean#L481-L493) from core?